src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborDataItem.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Security.Cryptography.Encoding.Tests.Cbor
+{
+    internal enum CborMajorType : byte
+    {
+        UnsignedInteger = 0,
+        NegativeInteger = 1,
+        ByteString = 2,
+        Utf8String = 3,
+        Array = 4,
+        Map = 5,
+        Tag = 6,
+        Special = 7,
+    }
+
+    /// Represents the Cbor Data item initial byte structure
+    internal struct CborDataItem
+    {
+        private const byte AdditionalInformationMask = 0b000_11111;
+        public readonly byte InitialByte;
+
+        public CborDataItem(CborMajorType majorType, byte additionalInfo)
+        {
+            Debug.Assert((byte)majorType < 8, "CBOR Major Type is out of range");
+            Debug.Assert((byte)additionalInfo < 32, "CBOR initial byte additional info is out of range");
```

Comment:
Assuming this type is expected to be public at some point, these should be checks, not asserts.

If it's supposed to always be internal, declare the ctor internal :smile:.

Reply:
It's really meant to be internal.

Reply:
I'm not sure how using `internal` vs `public` members on an internal type changes things in this case. Is it a style issue?

Reply:
If a member is declared as `public` I assume you mean that it will be public when the feature goes for API review; if it says `internal` I assume it'll stay internal.

The types are internal, but the members I'm looking at with an eye toward their final implementation.